udp-native/src/main/java/org/ballerinalang/stdlib/udp/SelectorManager.java

```diff
@@ -215,18 +216,11 @@ public void invokeRead(int socketHashId) {
     private void readUdpSocket(SocketReader socketReader, ReadPendingCallback callback) {
         DatagramChannel channel = (DatagramChannel) socketReader.getSocketService().getSocketChannel();
         try {
-            ByteBuffer buffer = createBuffer(callback, channel);
-            final InetSocketAddress remoteAddress = (InetSocketAddress) channel.receive(buffer);
-            callback.resetTimeout();
-            final int bufferPosition = buffer.position();
-            callback.updateCurrentLength(bufferPosition);
-            // Re-register for read ready events.
-            socketReader.getSelectionKey().interestOps(OP_READ);
-            selector.wakeup();
-            byte[] bytes = SocketUtils
-                    .getByteArrayFromByteBuffer(callback.getBuffer() == null ? buffer : callback.getBuffer());
-            callback.getFuture().complete(createUdpSocketReturnValue(callback, bytes, remoteAddress));
-            callback.cancelTimeout();
+            if (callback.getCalledBy() == SocketConstants.CallFrom.CONNECTIONLESS_CLIENT) {
+                processConnectionLessClientReceive(socketReader, callback, channel);
+            } else if (callback.getCalledBy() == SocketConstants.CallFrom.CONNECT_CLIENT) {
```

Comment:
If the calledBy can have only either of two values, no need for the else-if. Only else is sufficient I suppose.

Reply:
+1, LISTENER type will be added later when implementing the listener